icepyx/core/APIformatting.py

```diff
@@ -36,10 +36,6 @@ def _fmt_temporal(start, end, key):
 
     assert isinstance(start, dt.datetime)
     assert isinstance(end, dt.datetime)
-    assert key in [
-        "time",
-        "temporal",
-    ], "An invalid time key was submitted for formatting."
```

Comment:
This logic (is `key` `"time"` or `"temporal"`?) is repeated just below, so I moved the error statement into that conditional.

In general, assertions should not be used in "the code" (use only in the tests) unless you specifically want the assertions to be controllable at runtime. `python -O` will disable all assertions like `-W` disables warnings. So when I moved this logic I also changed it to raise a `RuntimeError`.